src/server/status_server/mod.rs

```diff
@@ -308,6 +312,108 @@ where
         })
     }
 
+    async fn get_cmdline(_req: Request<Body>) -> hyper::Result<Response<Body>> {
+        let args = args().into_iter().fold(String::new(), |mut a, b| {
+            a.push_str(&b);
+            a.push_str("\x00");
+            a
+        });
+        let response = Response::builder()
+            .header("Content-Type", mime::TEXT_PLAIN.to_string())
+            .header("X-Content-Type-Options", "nosniff")
+            .body(args.into())
+            .unwrap();
+        Ok(response)
+    }
+
+    // The request and response format follows pprof remote server https://gperftools.github.io/gperftools/pprof_remote_servers.html
+    // Here is the go pprof implementation https://github.com/golang/go/blob/3857a89e7eb872fa22d569e70b7e076bec74ebbb/src/net/http/pprof/pprof.go#L191
+    async fn get_symbol(req: Request<Body>) -> hyper::Result<Response<Body>> {
+        let file = std::fs::File::open(current_exe().unwrap()).unwrap(); // TODO: handle error 
+        let map = unsafe { memmap2::Mmap::map(&file).unwrap() };
+        let object = object::File::parse(&*map).unwrap();
```

Comment:
Can we get symbol using `backtrace::resolve`? TiKV already caches addr2line in backtrace. https://github.com/rust-lang/backtrace-rs/blob/a3406f93f3e6d91e05525c671fa694a0aea22c82/src/symbolize/gimli.rs#L261